### DISCUSSION TITLE
PMM Using too much ram

### DISCUSSION THREAD

**Discussion by thedinz** [meisnate12/Plex-Meta-Manager#859] [Q&A]
Maybe this should go under issues but i wasn't sure.

I have been noticing that the ram usage on my UNraid machine as been higher than normal the past few weeks. The behavior was weird to, i would restart all containers or the whole server, and my ram usage (for context we are talking while idle) would be what it always as been but over the next few days, it would tick up and up, and up slowly and stay there. NOT go up because tasks were being run and then back down, just slowly go up and stay up.

Anyway, i think I have pinpointed PMM as the issue, just now the ram for PMM while sitting idle was 2.5GB, far more then any other container. I reboot PMM and the usage went down to what it should be, now over the next few days, this will grow. Again, not while PMM is running tasks but at idle. Before rebooting to the container I checked the logs and it said its task has finished successfully at 6AM this was 10AM when i have noticed the ram usage and rebooted the container.

Just looking for some help here, I do believe this is new as I have been running PMM for a few months now and the ram issue is somewhat recent.

**Comment by thedinz** [comment]
Still happening...

**Reply by meisnate12** [reply]
sorry im trying to get 1.17.0 which i think may solve the issue

**Reply by gitthangbaby** [reply]
It kills my powerful NAS easily:
_[python3] stopped running because the system is out of memory._
i'm looking for ways how to make it run quicker, even if I do just one action, it takes so much time. And do the same task over and over every day, i'm sure 99% actions could be skipped.

**Reply by YozoraXCII** [reply]
Do you have a log file and we can see what exactly Kometa is doing to see if any imrpovements can be made?

**Reply by gitthangbaby** [reply]
> Do you have a log file and we can see what exactly Kometa is doing to see if any imrpovements can be made?

Hi
i checked the occurences and it's pretty random (python3.8 + 3.12 tested), in any type of execution. i run --operations-only, --overlays-only daily which takes 1 hour for ~1000 items, and at weekend i run all which takes hours. but recently short execution takes 4-8hours, with all NAS components breaking, snapshots, cameras etc.. failing because of memory hog.

```
Series Library Operations
<...>
Processing 1883 S01E09
<frozen, system kills python3>
```

```
Series Library Operations
<...>
Processing Ash vs Evil Dead S03E09
<frozen, system kills python3>
```

```
Running HBO Max Shows Collection
<...>
Convert Warning: No TVDb ID Found for TMDb ID: 220345
^C|====================================================================================================|
|                                           Exiting Kometa                                           |
|====================================================================================================|
<it is actually very slowly progressing but NAS is dead, i do CTRL-C which triggers exit which doesn't happen, i then kill it and it takes an hour to finally exit>
```

I have no new movies for months, and i don't understand what's the need to do same operation over and over, as i'm fetching just MDB commonsense ratings in operations, and overlays stay the same (well maybe some user ratings change microscopically but I really don't need to have them updated). What's changes in e.g. Oscar collections, I also don't know. I wish the existing items were skipped globally, or refreshed once a month only and everything was cached.

**Reply by gitthangbaby** [reply]
i've added +50% (8GB) memory (spike 3 on chart) and it solved the memory hog issue:
<img width="913" height="369" alt="image" src="https://github.com/user-attachments/assets/491eec76-bfef-40b0-9eab-9054509a8237" />
